OrbitLinuxTracing/PerfEventRingBuffer.cpp

```diff
@@ -85,32 +85,36 @@ bool PerfEventRingBuffer::HasNewData() {
 }
 
 void PerfEventRingBuffer::ReadHeader(perf_event_header* header) {
-  assert(IsOpen());
   ReadAtTail(reinterpret_cast<uint8_t*>(header), sizeof(perf_event_header));
   assert(header->type != 0);
   assert(metadata_page_->data_tail + header->size <= metadata_page_->data_head);
 }
 
 void PerfEventRingBuffer::SkipRecord(const perf_event_header& header) {
-  assert(IsOpen());
   // Write back how far we read from the buffer.
   metadata_page_->data_tail += header.size;
 }
 
 void PerfEventRingBuffer::ConsumeRecord(const perf_event_header& header,
                                         void* record) {
-  assert(IsOpen());
   ReadAtTail(static_cast<uint8_t*>(record), header.size);
   SkipRecord(header);
 }
 
+void PerfEventRingBuffer::ReadRawAtOffset(void* dest, uint64_t offset,
+                                          uint64_t count) {
+  ReadAtOffsetFromTail(static_cast<uint8_t*>(dest), offset, count);
+}
+
 void PerfEventRingBuffer::ReadAtTail(uint8_t* dest, uint64_t count) {
   return ReadAtOffsetFromTail(dest, 0, count);
 }
 
 void PerfEventRingBuffer::ReadAtOffsetFromTail(uint8_t* dest,
                                                uint64_t offset_from_tail,
                                                uint64_t count) {
+  assert(IsOpen());
```

Comment:
Replace this with CHECK?

Reply:
So, CHECK is in OrbitCore. Should I define another CHECK in this module?

Reply:
we should create OrbitBase at some point...

Reply:
+1 There are other basic debugging/dev things that are missing here that are present in OrbitCore.